esp-hal/Cargo.toml

```diff
@@ -20,6 +20,7 @@ bitfield                 = "0.15.0"
 cfg-if                   = "1.0.0"
 critical-section         = "1.1.2"
 defmt                    = { version = "0.3.6", optional = true }
+delegate                 = "0.12.0"
```

Comment:
yes - another dependency - but it only depends on things we already depend on and I guess it will be useful in other places in future. Technically we could get rid of it (by replacing it with some more boiler-plate code)